operators/src/source/gdal_source/mod.rs

```diff
@@ -653,7 +655,10 @@ where
             tiling_strategy.tile_size_in_pixels,
         );
 
-        Ok(filled_stream.boxed())
+        // TODO: automatically wrap in all operators. Doing it like for the statistics opeator is not possible
+        //       because the cancellation should be per query and not per QueryProcessor. Thus the cancellation
+        //       must be part of the QueryContext and not the ExecutionContext.
+        Ok(ctx.valve().wrap(filled_stream).boxed())
```

Comment:
What is the difference between this and the functionality within `tokio`?
Moreover, could we have some custom struct that wraps this so that we are not dependent on stream-cancel? It seems this isn't being updated for over a year. We should lower the dependencies since it is within our core.

Reply:
the difference is that you can re-use the valve. You can wrap multiple streams with it and abort them all at the same time. I think we need this.

Reply:
You omitted the second question, but this is super important.